idaes/models/properties/modular_properties/base/generic_property.py

```diff
@@ -3715,7 +3818,11 @@ def _therm_cond_phase(self):
         try:
 
             def rule_therm_cond_phase(b, p):
-                return get_phase_method(b, "therm_cond_phase", p)(b, p)
+                try:
+                    return get_phase_method(b, "therm_cond_phase", p)(b, p)
+                except GenericPropertyPackageError:
+                    # Handle case where thermal conductivity isn't defined for a phase
```

Comment:
I am a little bit cautious about this - what are the implications of having this missing, and what error message would a user receive if they tried accessing a missing entry? This is probably the right thing to do, but what happens in a unit model if it tries to get a missing thermal conductivity?

Reply:
Yeah - that error is not going to be helpful to the general user, especially one who is taking a property definition and unit model they did not write themselves.

We have a problem here that this is general purpose code, and so it needs to work gracefully in all circumstances, which this unfortunately does not. I'll need to think more about this, but I definitely do not like leaving these as `Expression.Skip`.

Reply:
I am still concerned about `Expression.Skip` in these cases - for the phase-component properties I think we are OK as we are making those private attributes now, but for these public facing ones I think we need to be more careful.

I think we either need to provide a meaningful value for that property in all phases, or we need to make sure that a meaningful error is returned (which I don't think is possible as that occurs outside the property package). Thus, my initial inclination is that we need to insist that users define a method for all phases, even in cases where they know for sure it won;t be used (they can always assign a constant value). This also helps make it clear to any downstream users, as they can see from the config that a constant value is used (whereas an absent value could be overlooked).

The alternative is to have some way of making sure the user gets a useful message - i.e. PropertyNotSupportedError, and probably a logger warning when the Expression is skipped as well. However, I am not sure how we would actually catch that.